paypal/driver.py

```diff
@@ -273,21 +294,12 @@ def DoExpressCheckoutPayment(
             'PAYERID': payerid,
         }
         parameters.update(kwargs)
-        query_string = self.signature + urllib.urlencode(parameters)
-        response = urllib2.urlopen(self.NVP_API_ENDPOINT, query_string).read()
-        response_tokens = {}
-        for token in response.split('&'):
-            response_tokens[token.split("=")[0]] = token.split("=")[1]
-        for key in response_tokens.keys():
-            response_tokens[key] = urllib.unquote(response_tokens[key])
+        state = self._nvp_request(parameters)
 
-        state = self._get_value_from_qs(response_tokens, "ACK")
-        self.response = response_tokens
-        self.api_response = response_tokens
-        if not state in ["Success", "SuccessWithWarning"]:
+        if state not in ["Success", "SuccessWithWarning"]:
             self.doexpresscheckoutpaymenterror = GENERIC_PAYMENT_ERROR
             self.apierror = self._get_value_from_qs(
-                response_tokens, "L_LONGMESSAGE0"
+                self.response, "L_LONGMESSAGE0"
```

Comment:
refactoring common functionality into _nvp_request means we don't have a response_tokens variable, but self.response is set to response_tokens anyway see L141